### ISSUE TITLE
-Wsometimes-uninitialized in arch/arm/mach-omap1/pm.c

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1324]
via omap1_defconfig on linux-next:
```
arch/arm/mach-omap1/pm.c:656:11: warning: variable 'irq' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
        else if (cpu_is_omap16xx())
                 ^~~~~~~~~~~~~~~~~
./arch/arm/mach-omap1/include/mach/soc.h:123:30: note: expanded from macro 'cpu_is_omap16xx'
#  define cpu_is_omap16xx()             is_omap16xx()
                                        ^~~~~~~~~~~~~
arch/arm/mach-omap1/pm.c:658:18: note: uninitialized use occurs here
        if (request_irq(irq, omap_wakeup_interrupt, 0, "peripheral wakeup",
                        ^~~
arch/arm/mach-omap1/pm.c:656:7: note: remove the 'if' if its condition is always true
        else if (cpu_is_omap16xx())
             ^~~~~~~~~~~~~~~~~~~~~~
arch/arm/mach-omap1/pm.c:611:9: note  CC      ipc/syscall.o
  CC      mm/vmscan.o
: initialize the variable 'irq' to silence this warning
        int irq;
               ^
                = 0
```

**Comment by nickdesaulniers** [comment]
submitted: https://lore.kernel.org/lkml/20210401161127.8942-1-maciej.falkowski9@gmail.com/

**Comment by m-falkowski** [comment]
Patch accepted: https://lore.kernel.org/lkml/YKNfgRWM9MK1MY9O@atomide.com/

**Comment by nickdesaulniers** [comment]
fixed: 3c4e0147c269738a19c7d70cd32395600bcc0714
